apps/server/src/modules/oauth/controller/oauth-sso.controller.ts

```diff
@@ -18,9 +19,15 @@ export class OauthSSOController {
 		@Query() query: AuthorizationParams,
 		@Res() res: Response,
 		@Param('systemid', ParseObjectIdPipe) systemid: string
-	): Promise<unknown> {
+	): Promise<void> {
 		const oauthResponse = await this.oauthUc.startOauth(query, systemid);
-		res.cookie('jwt', oauthResponse.jwt ? oauthResponse.jwt : '');
-		return res.redirect(oauthResponse.redirect ? oauthResponse.redirect : '');
+		const cookieDefaultOptions: CookieOptions = {
```

Comment:
Can you please check following code snippets?

I am on writing documentation on github for configuration in ticket BC-2347. But is not pushed or finished right now.
But we want to replace configuration in future with the nestjs solution for it and remove Configuration from commons. It is helpful for easyer switch to prepare it now on each places. It is a little more overhead, but reduce a lot of work in future. I find that make it easyer to mock values in test too.

For this we add a interface in each modul what envirment settings are needed. All this interfaces moved to top and combined to a "all what you need as configuration to start this application" interface.

https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/server.config.ts#L18

This configuration is passed to a global aviable nestjs configuration module and service. 
https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/server.module.ts#L36

It can be used in different ways. But for this place it can be injected and reduce by your own interface like on this place.
https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/user/uc/user.uc.ts#L32

Maybe for sameSite a enum is helpful in future.

Reply:
That sounds nice. Thanks for the heads up. We will consider this in our next implementations. As this is just a bug fix we think this is not the right place to implement that for the first time. When you are finished (a brownbag session would be really nice :) ) we could create a refactoring ticket and fix that in our code.

Reply:
ok